### DISCUSSION TITLE
Error while linting within VS Code.

### DISCUSSION THREAD

**Discussion by seRRaloR** [stoplightio/spectral#2756] [Q&A]
Hi,

I'm receiving the following error while linting any OAS 3.0 file within VS Code:
> Spectral: An error occurred while validating file:_my_file_being_linted. Please see the 'Spectral' output channel for details.

And in the Spectral output channel, this is what I have:
![image](https://github.com/user-attachments/assets/03a827c7-55be-4733-94ce-7269ac5eca86)

These are my Spectral settings:
![image](https://github.com/user-attachments/assets/40221cbf-a4dc-46a8-b232-a4c91ed8455e)

I'm using VS Code Versión 1.94.2 and Spectral v1.1.4. plugin.

I don't know if I should attach the ruleset file, but if so, please tell.

It's giving me this error no matter what file being linted.

There are some teammates of mine that have it working properly with the same ruleset, so I gues it's not a ruleset issue.

Any ideas or suggestions?

Thank you in advance.

**Comment by frankkilcommins** [comment]
The v1.1.4 of the extension is working OK for me with OAS 3.0.* descriptions. 

What's the contents of your ruleset JS file? Do you get the same issue if using the default .spectral.yaml at the root of your project?

**Reply by seRRaloR** [reply]
Excuse my ignorance but, where can I find that defaul ruleset.

As I've mencioned, I'm working with de VS Code plugin, and it's installed in **%USERPROFILE%\.vscode\extensions\stoplight.spectral-1.1.4** and there is not such file in this path.


Thanks in advance.

**Reply by frankkilcommins** [reply]
No worries at all. By _root of your project_, I mean what ever folder/workspace is active for you in VS Code. See example below:
<img width="364" alt="image" src="https://github.com/user-attachments/assets/fdc7235d-a3d1-4239-8996-584a0d36b8e9" />

Here's the contents of the `.spectral.yaml` file (in this case I want to leverage the core ruleset for OAS and Arazzo):
```
extends: 
  - "spectral:oas"
  - "spectral:arazzo"
```

Also as an FYI, here is what I had in my spectral.js file then verifying the OAS 3.0.* was working using a specific location for a JS ruleset file:
```
import { truthy, undefined as pattern, schema } from "@stoplight/spectral-functions";
import { oas3 } from "@stoplight/spectral-formats";
```

**Reply by seRRaloR** [reply]
Sorry...

I don't know why but I've removed my custom ruleset from settings and (don't know how the plugin has come to default ruleset) but it's working fine.
